crates/matrix-sdk/src/event_handler.rs

```diff
@@ -83,6 +84,25 @@ pub trait SyncEvent {
     const ID: (EventKind, &'static str);
 }
 
+pub(crate) struct EventHandlerWrapper {
+    pub handler_function: Box<EventHandlerFn>,
+    pub handle: EventHandlerHandle,
+}
+
+#[derive(Debug, Clone)]
+/// Handle to remove a registered event handler by passing it to
+/// [`Client::remove_event_handler`].
+pub struct EventHandlerHandle {
+    pub(crate) id: (EventKind, &'static str),
+    pub(crate) addr: u64,
```

Comment:
The `addr` name no longer makes sense. How about this (the current `ID` tuple is removed in #878 anyways):

```suggestion
    pub(crate) ev_kind: EventKind,
    pub(crate) ev_type: &'static str,
    pub(crate) id: u64,
```

Reply:
Since we need a tuple of `ev_kind` *and* `ev_type` and never really only one of them, I will save both as a tuple called `ev_id`, whereas the other id will be called `handler_id` for it identifies the handler and not the handle.

Reply:
Do we really need them to be "tupled" together?

Reply:
I can deconstruct them, then put them each in a field and put them together afterwards, but I don't see why that's more useful?

Reply:
Hm, okay. I guess because the tuple is currently used as the map key, it's okay. I intend to change that soon, but can "un-tuple" the fields then.